cmd/cluster-agent/api/v1/install.go

```diff
@@ -35,10 +36,47 @@ func Install(r *mux.Router, sc clusteragent.ServerContext) {
 	r.HandleFunc("/tags/pod/{nodeName}", getPodMetadataForNode).Methods("GET")
 	r.HandleFunc("/tags/pod", getAllMetadata).Methods("GET")
 	r.HandleFunc("/tags/node/{nodeName}", getNodeMetadata).Methods("GET")
+	r.HandleFunc("/tags/cf/apps/{nodeName}", getCFAppsMetadataForNode).Methods("GET")
 	installClusterCheckEndpoints(r, sc)
 	installEndpointsCheckEndpoints(r, sc)
 }
 
+// getCFAppsMetadataForNode is only used when the node agent hits the DCA for the list of cloudfoundry applications tags
+// It return a list of tags for each application that can be directly used in the tagger
+func getCFAppsMetadataForNode(w http.ResponseWriter, r *http.Request) {
```

Comment:
it's a bit confusing to see cf and k8s handlers in the same file, can we at least move them to separate files? e.g `kubernetes_metadata.go` and `cloudfoundry_metatada.go` with an `install` in every file and we call them in  `Install` the same way we call `installClusterCheckEndpoints` and `installEndpointsCheckEndpoints` wdyt?
we can also limit the installed endpoints depending on the environment, there is no need to expose k8s tag endpoints when cf enabled and vice versa - this could reduce the security risks as these endpoints are publicly accessible

Reply:
How do I limit the installed endpoints ?

Reply:
we can check the config of the agent to know which mode is activated, although this makes wonder why we're keeping this pkg under `cluster-agent` - maybe we need to move it in a common package and both cluster agent can plug to it